### ISSUE TITLE
Plain MenuItem on MenuBar is misaligned

### ISSUE THREAD

**Issue by mnesarco** [ocornut/imgui#4538]
```
Dear ImGui 1.85 WIP (18412)
--------------------------------
sizeof(size_t): 8, sizeof(ImDrawIdx): 2, sizeof(ImDrawVert): 20
define: __cplusplus=202002
define: __linux__
define: __GNUC__=4
define: __clang_version__=12.0.1 
define: IMGUI_HAS_VIEWPORT
define: IMGUI_HAS_DOCK
--------------------------------
io.BackendPlatformName: imgui_impl_glfw
io.BackendRendererName: imgui_impl_opengl3
io.ConfigFlags: 0x00000000
io.ConfigViewportsNoDecoration
io.ConfigInputTextCursorBlink
io.ConfigWindowsResizeFromEdges
io.ConfigMemoryCompactTimer = 60.0
io.BackendFlags: 0x0000140E
 HasMouseCursors
 HasSetMousePos
 PlatformHasViewports
 RendererHasVtxOffset
 RendererHasViewports
--------------------------------
io.Fonts: 1 fonts, Flags: 0x00000000, TexSize: 512,64
io.DisplaySize: 1920.00,1023.00
io.DisplayFramebufferScale: 1.00,1.00
--------------------------------
style.WindowPadding: 8.00,8.00
style.WindowBorderSize: 1.00
style.FramePadding: 4.00,3.00
style.FrameRounding: 0.00
style.FrameBorderSize: 0.00
style.ItemSpacing: 8.00,4.00
style.ItemInnerSpacing: 4.00,4.00
```

**Version/Branch of Dear ImGui:**

Version: Dear ImGui 1.85 WIP (18412)
Branch: docking

**Back-end/Renderer/Compiler/OS**

Back-ends: 
imgui_impl_glfw
imgui_impl_opengl3

Compiler: clang 12.0
Operating System: Linux Mint 19.3

**My Issue/Question:**

When using MenuItems directly on a MenuBar, they are not aligned with Menus.

**Screenshots/Video**

![screenshot_11](https://user-images.githubusercontent.com/1087317/133148745-efbe3d54-dc8d-4f7d-8241-4a96de1bb101.png)

**Standalone, minimal, complete and verifiable example:** _(see https://github.com/ocornut/imgui/issues/2261)_

```cpp
// Here's some code anyone can copy and paste to reproduce your issue

if (ImGui::BeginMainMenuBar())
{   
    if (ImGui::BeginMenu("item1"))
    {
        // ....
        ImGui::EndMenu();
    }

    if (ImGui::MenuItem("item2"))
    {
        // .....
    }

    ImGui::EndMainMenuBar();
}
```

**Comment by ocornut** [comment]
Thank you for reporting this. Broken by f8fae022. I'll work on a fix.

**Comment by ocornut** [comment]
Pushed fix 7b8bc86, sorry for the issue.
This type of bugs are not well supported by our regression testing framework yet, but we have some plans to work toward that (queued work on a test for this right now).
